docs/design/bundler.md

```diff
@@ -449,6 +453,58 @@ because then it's trying to enforce further checks as outlined in the
 Therefore, we additionally need to enforce both `BUNDLE_DISABLE_LOCAL_BRANCH_CHECK` and
 `BUNDLE_DISABLE_LOCAL_REVISION_CHECK`.
 
+#### Offline installation using deployment mode
+Deployment mode is a way of vendoring one's code along with the dependencies.
+The most important bit about the deployment mode based on the official
+[docs](https://www.bundler.cn/man/bundle-install.1.html#DEPLOYMENT-MODE)
+
+>Gems are installed to vendor/bundle not your default system location.
+>
+>In development, it's convenient to share the gems used in your application with other applications
+>and other scripts that run on the system.
+>
+>In deployment, isolation is a more important default. In addition, the user deploying the
+>application may not have permission to install gems to the system, or the web server may not have
+>permission to read them.
+>As a result, bundle install --deployment installs gems to the vendor/bundle directory in the
+>application. This may be overridden using the --path option.
+
+is that this creates a local `vendor/bundle` directory local to the application repository instead
+of using a system-wide location during the install. What that means in practice is that regardless
+of the Rubygems ecosystem recommendations we're enforcing a particular way of installing
+applications onto our whole user base. If they wish to override it, they need to make changes to
+their build recipes and explicitly by setting the `BUNDLE_PATH` configuration option and point it
+to a system location. That is official the only way to tell bundler to skip creating the local
+`vendor/bundle` directory using the deployment mode.
+
+Trying not to use the deployment mode (without making use of `--local` on the user side) leads to
```

Comment:
Structural nitpick: the rest of this section belongs to a separate section, `Typical problems which arise when trying to enforce offline installation`. I would have also swapped this subsection with `Offline installation using deployment mode`

Reply:
> I would have also swapped this subsection with Offline installation using deployment mode

What do you mean? Like only use that bit of content and drop the rest?

Reply:
1. Add a new header `Typical problems which arise when trying to enforce offline installation` on L479;
2. Cut L479-L506 and paste after L455 (i.e. right before `Offline installation...`);
3. Keep everything else.

The contents itself is fine, it is just about ordering of subsections. Also the suggested subsection name is verbose and clunky, something more succinct would be much better, I just could not come up with a better one yet. This is not a fundamental blocker, but I think it will help the reader a bit.

Reply:
Adjusted, although a bit differently than suggested @a-ovchinnikov. Merely cutting and pasting the suggested content to a location earlier in the document didn't feel (contextually) right to me and so I reworded the paragraph slighly and added a `Offline installation pitfalls` section to host it, let me know if you agree.